glean-core/src/upload/directory.rs

```diff
@@ -17,6 +17,17 @@ use uuid::Uuid;
 use super::request::HeaderMap;
 use crate::{DELETION_REQUEST_PINGS_DIRECTORY, PENDING_PINGS_DIRECTORY};
 
+// The maximum number of files we process.
+// If we detect more files in the pending pings directory they are removed without processing.
+// Usually we read all files into memory to sort them and then discard the oldest ones if
+// necessary.
+// However if a user has too many files that's very costly (in both processing time and
+// memory usage), so we add a high watermark.
+// This means there's now a case where we can drop arbitrary data.
+// However if a user has _that many pending pings_ some things might have gone awry and the
+// data is less important than keeping the user's application running smoothly.
+const HIGH_WATERMARK_PENDING_PINGS: u64 = super::policy::MAX_PENDING_PINGS_COUNT * 4;
```

Comment:
Any reasoning behind the `* 4` we can document along with this? In case we want/need/have to change this in the future, it would be nice to understand the current value.

Reply:
I picked a number out of the hat. Open to discussion (`MAX_PEDING_PINGS_COUNT` is 250, so this would be 1000, that sounded like a nice round number, but that's all the thinking that went into it)